src/stan/mcmc/hmc/nuts_classic/adapt_dense_e_nuts_classic.hpp

```diff
@@ -0,0 +1,55 @@
+#ifndef STAN_MCMC_HMC_NUTS_ADAPT_DENSE_E_NUTS_CLASSIC_HPP
```

Comment:
Is the point of `nuts_classic` to keep the old implementation?  And if so, do we really need both?  I don't care about having backward compatibility at the underlying algorithm level.